runtime/src/in_mem_accounts_index.rs

```diff
@@ -648,34 +683,38 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
 
         // this becomes inclusive - that is ok - we are just roughly holding a range of items.
         // inclusive is bigger than exclusive so we may hold 1 extra item worst case
-        let inclusive_range = Some(start..=end);
+        let inclusive_range = start..=end;
         let mut ranges = self.cache_ranges_held.write().unwrap();
+        let mut already_held = false;
         if start_holding {
```

Comment:
How would you feel about created separate functions for the `start_holding == true` and the `start_holding == false` cases? Then calling them in their appropriate `if/else` blocks.

Reply:
I think I may end up refactoring this some more soon. atm, this works and is better. I think I may move the whole held ranges behavior into its own struct and file.